### ISSUE TITLE
Possible to use cordova-plugin-opentok within 'ionic serve' and/or Ionic View app?

### ISSUE THREAD

**Issue by ben-i2x** [opentok/cordova-plugin-opentok#62]
Howdy -

I've added the cordova-plugin-opentok to my project, and when I build and deploy the app to my Android device it works well.

When I run: 'ionic serve' and use the local browser instance, I get an error that OT is undefined.  If I manually copy opentok.js into the top level 'www', I get an error about Cordova not being defined.  From poking around on the web, I see that this isn't surprising: cordova plugins don't appear to function in a 'ionic serve' context.

What's really vexing is that I'm running into a similar issue using the Ionic View App. I can push a build to the ionic pro dashboard, and can access the newly built app in Ionic View on my Android phone.

However, when I browse to a page that uses the top level variable OT, I get an error that it's not defined.

Is this plugin simply not going to work in both a 'ionic serve' and within the Ionic View app context?  Or, did I mess up my setup somewhere along the way?

What info I can I provide to make debugging this possible?

My Environment:

```
$ ionic info

cli packages: (C:\Users\ben\AppData\Roaming\npm\node_modules)

    @ionic/cli-utils  : 1.19.2
    ionic (Ionic CLI) : 3.20.0

global packages:

    cordova (Cordova CLI) : 7.1.0

local packages:

    @ionic/app-scripts : 3.1.8
    Cordova Platforms  : android 7.0.0
    Ionic Framework    : ionic-angular 3.9.2

System:

    Android SDK Tools : 25.2.5
    Node              : v6.11.4
    npm               : 5.6.0
    OS                : Windows 10

Environment Variables:

    ANDROID_HOME : c:/tools/android

Misc:

    backend : pro

$ ionic cordova plugins
> cordova plugin ls
com.tokbox.cordova.opentok 3.1.2 "OpenTokCordovaPlugin"
cordova-plugin-device 2.0.1 "Device"
cordova-plugin-ionic-keyboard 2.0.5 "cordova-plugin-ionic-keyboard"
cordova-plugin-ionic-webview 1.1.16 "cordova-plugin-ionic-webview"
cordova-plugin-screen-orientation 3.0.1 "Screen Orientation"
cordova-plugin-splashscreen 5.0.2 "Splashscreen"
cordova-plugin-whitelist 1.3.3 "Whitelist"
es6-promise-plugin 4.2.2 "Promise"


```

**Comment by msach22** [comment]
@ben-i2x thanks for filing an issue. The reason that `OT` is undefined is b/c the plugin currently only supports iOS and Android. For web, you should use the following node module: `https://www.npmjs.com/package/@opentok/client`. In the near future, we will be releasing an update to the plugin that will automatically download the OpenTok JS SDK if the platform is the browser. 

Hope this helps!

**Comment by ben-i2x** [comment]
@msach22  - thanks for the clarification and the work-around suggestion.

I get that I can run:

```npm install --save @opentok/client```

to add this module to my project.

Any chance you could give me an example of how I can import this into an Angular 2 / Ionic page?

Do I need to be concerned with the OT variable from @opentok/client clashing with the cordova-plugin-opentok variable?

Thanks again for the assistance, I'm still wrapping my head around the all the various layers that go into an Ionic app.

**Comment by mark-veenstra** [comment]
With version 1 of Ionic the plugins are not initialized during Ionic serve. Don't know if they do with later versions.

But you could build the browser platform and then serve the platform/browser/www directory.
This is how we do it anyways.

You need to wait for the 3.2.0 release since it hence the platform browser

**Comment by msach22** [comment]
@mark-veenstra as far as I know, Ionic 2 behaves the same b/c in my app, the plugin is not initialized until `ionic serve`

@ben-i2x We've released Cordova 3.2.0 so please use the updated plugin if you would like to use the browser platform.

**Comment by ben-i2x** [comment]
Thanks @msach22, I'll give that a try.
